lib/SIL/IR/SILInstruction.cpp

```diff
@@ -1002,6 +1002,22 @@ SILInstruction::MemoryBehavior SILInstruction::getMemoryBehavior() const {
     }
   }
 
+  if (auto *ga = dyn_cast<GlobalAddrInst>(this)) {
+    // Global variables with resilient types might be allocated into a buffer
+    // and not statically in the data segment.
+    // In this case, the global_addr depends on alloc_global being executed
+    // first. We model this by letting global_addr have a side effect.
+    // It prevents e.g. LICM to move a global_addr out of a loop while keeping
+    // the alloc_global inside the loop.
+    SILModule &M = ga->getFunction()->getModule();
+    auto expansion = TypeExpansionContext::maximal(M.getAssociatedContext(),
```

Comment:
Since IRGen makes a similar calculation when lowering the instruction to decide if it should allocate the box or not, should we factor it out somewhere?

Reply:
IRGen uses the TypeInfo machinery for this, which is only available in IRgen